### PR TITLE
feat: add search api endpoint

### PR DESCRIPTION
**Pull request by fgouteroux**
Hello,

I would add the search api endpoint, it could be useful for [terraform-provider-ns1](https://github.com/ns1-terraform/terraform-provider-ns1) to perform some additional validations.

It should resolved the issue https://github.com/ns1/ns1-go/issues/152

### CODE REVIEW

**Comment by fgouteroux** [comment]
Hello, any updates, review ?

File: rest/search.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	"net/http"
+
+	"gopkg.in/ns1/ns1-go.v2/rest/model/dns"
+)
+
+// RecordSearchService handles 'dns/record/search' endpoint.
+type RecordSearchService service
+
+// Find takes query parameters and returns matching DNS records.
+func (s *RecordSearchService) Find(params string) (*dns.SearchResult, *http.Response, error) {
```
Comment (dmavrommatis) [inline rest/search.go]:
```suggestion
func (s *RecordSearchService) Search(params string) (*dns.SearchResult, *http.Response, error) {
```

File: rest/search.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	}
+
+	return &r, resp, nil
+}
+
+// ZoneSearchService handles 'dns/zone/search' endpoint.
+type ZoneSearchService service
+
+// Find takes query parameters and returns matching DNS zones.
+func (s *ZoneSearchService) Find(params string) (*dns.SearchResult, *http.Response, error) {
```
Comment (dmavrommatis) [inline rest/search.go]:
```suggestion
func (s *ZoneSearchService) Search(params string) (*dns.SearchResult, *http.Response, error) {
```

**Review by dmavrommatis — CHANGES_REQUESTED** [review]
Some small changes and it should be good to go!

**Review by dmavrommatis — APPROVED** [review]
LGTM :tada:

**Review by ponbiki — APPROVED** [review]

**Comment by fgouteroux** [comment]
Hello, still not merged ?

**Pull request merged**